components/modelregistry/modelregistry.go

```diff
@@ -180,6 +187,38 @@ func (m *ModelRegistry) ReconcileComponent(ctx context.Context, cli client.Clien
 	return nil
 }
 
+// removeUnusedKubeRbacProxy removes older kube-rbac-proxy container in mr operator deployment.
+// This is required because patching a deployment using apply-patch doesn't remove older containers.
+// See: https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#notes-on-the-strategic-merge-patch
+func removeUnusedKubeRbacProxy(ctx context.Context, cli client.Client, name string, namespace string) error {
```

Comment:
Good question. I'm not sure how the one time upgrade logic worked, so I went with the simplest approach keeping the changes in the model registry component. But if there is a way to detect the version upgrade and run this once, that's better. Please feel free to modify this PR to do it, since I'm not that familiar with the upgrade logic.

Reply:
i made a new commit based on above converstation:
- during operator upgrade if it is ODH (no need for rhoai)
- and modelreg deployment exists
- and container in deployment is more than 1 
- remove the one with name kube-rbac-proxy